### PR TITLE
did validate_spec swagger schema agnostic

### PR DESCRIPTION
**Pull request by dutradda**
I have my own Swagger Schema Extension (with oneOf and anyOf support) and I need this changes to use my schema to validate under yours API.

### CODE REVIEW

**Comment by dutradda** [comment]
@analogue @jnb @prat0318

Can someone validate and merge my PR? This feature can be useful for other people too.

**Comment by dutradda** [comment]
@gnufied @rockdog @bobtfish 
hello!

Can someone give your opinion about this PR? Would be great if this feature were in the next version

File: swagger_spec_validator/validator20.py [hunk 1]
```diff
@@ -57,7 +57,9 @@ def validate_spec_url(spec_url):
     return validate_spec(load_json(spec_url), spec_url)
 
 
-def validate_spec(spec_dict, spec_url='', http_handlers=None):
+def validate_spec(spec_dict, spec_url='', http_handlers=None,
```
Comment (sjaensch) [inline swagger_spec_validator/validator20.py]: Maybe it makes more sense to provide a higher-level abstraction? Several possibilities come to mind, e.g. either passing in an arbitrary filesystem path, a file object from which we read or just the schema as JSON string. I wouldn't expose the internal `python package` + `relative_path` way of loading it as our API.

File: swagger_spec_validator/validator20.py [hunk 2]
```diff
@@ -69,6 +70,8 @@ def validate_spec(spec_dict, spec_url='', http_handlers=None):
         http client built into jsonschema's RefResolver is used. This
         is a mapping from uri scheme to a callable that takes a
         uri.
+    :param schema_pkg: package of the json schema file.
+    :param schema_path: package relative path of the json schema file.
```
Comment (sjaensch) [inline swagger_spec_validator/validator20.py]: These don't exist anymore. Please instead add a line documenting what `schema_dict` is supposed to be.

File: swagger_spec_validator/validator20.py [hunk 3]
```diff
@@ -91,10 +94,12 @@ def validate_spec(spec_dict, spec_url='', http_handlers=None):
 
 
 @wrap_exception
-def validate_json(spec_dict, schema_path, spec_url='', http_handlers=None):
+def validate_json(spec_dict, schema_path, spec_url='',
+                  http_handlers=None, schema=None):
     """Validate a json document against a json schema.
 
     :param spec_dict: json document in the form of a list or dict.
+    :param schema_pkg: package of the json schema file.
```
Comment (sjaensch) [inline swagger_spec_validator/validator20.py]: This is obsolete as well.

**Comment by diogodafiti** [comment]
@sjaensch Can you take a look in my PR, I had rebased it with the base branch.

**Comment by dutradda** [comment]
@sjaensch Can you take a look in my PR, I had rebased it with the base branch.

**Comment by dutradda** [comment]
@sjaensch Can you take a look on my PR?

**Comment by dutradda** [comment]
@sjaensch I tried to not broke the old api and tried to do it compatible. I just moved some functions to the args, so I can override this functions with another one. The functionally remains the same, because of this no extra tests are needed. I will check again for api breaks, but I think it is ok to merge.
I am sorry for my bad english.

**Comment by dutradda** [comment]
I thought about it and I understood your points better.
So I will change my PR for just a little things, and I will create a new package with the functions I need to use and I will import yours functions. Thank you!

**Pull request closed without merging**